src/directives/NgGrid.ts

```diff
@@ -387,7 +393,7 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 			}
 		}
 
-		if (this._autoResize) {
+		if (this._autoResize || (this._limitToScreen && this._centerToScreen)) {

```

Comment:
I don't think this check should be here. If anything, you should add an `|| this._centerToScreen` to the inner if statement above.

Reply:
Yes it could be as separate inner statement in the above if but then it would replicate those 3 lines of code so which ever you think is better. I was thinking of placing it inside the limitToScreen for better code readability, but wasn't sure if you like replicating that loop code.